public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php

```diff
@@ -427,6 +429,26 @@ function register_post_statuses() {
 	);
 }
 
+/**
+ * Do things when certain status transitions happen.
+ *
+ * @param string   $new_status
+ * @param string   $old_status
+ * @param \WP_Post $post
+ *
+ * @return void
+ */
+function status_transitions( $new_status, $old_status, $post ) {
+	if ( POST_TYPE !== get_post_type( $post ) ) {
+		return;
+	}
+
+	// If a pattern gets relisted, remove the reason that it was originally unlisted.
+	if ( UNLISTED_STATUS === $old_status && in_array( $new_status, array( 'publish', 'pending' ), true ) ) {
```

Comment:
I think you could say `&& UNLISTED_STATUS !== $new_status` here, since even if it moves to draft or possible spam, we don't need to keep the unlisted reason.

Reply:
👍 3ce923e